httomo/methods_database/query.py

```diff
@@ -0,0 +1,45 @@
+import yaml
+from pathlib import Path
+
+
+YAML_DIR = Path(__file__).parent / 'packages/'
+
+
+def get_method_info(module_path: str, method_name: str, attr: str):
+    """Get the information about the given method associated with `attr` that
+    is stored in the relevant YAML file in `httomo/methods_database/packages/`
+
+    Parameters
+    ----------
+    module_path : str
+        The full module path of the method, including the top-level package
+        name. Ie, `httomolib.misc.images.save_to_images`.
+
+    method_name : str
+        The name of the method function.
+
+    attr : str
+        The name of the piece of information about the method being requested
+        (for example, "pattern").
+
+    Returns
+    -------
+    TODO: Needs a "generic" type to represent anything that could be stored in
+    the YAML files in the methods database?
+        The requested piece of information about the method.
+    """
+    method_path = f"{module_path}.{method_name}"
+    split_method_path = method_path.split('.')
+    package_name = split_method_path[0]
+    yaml_info_path = Path(YAML_DIR, f"{package_name}.yaml")
+
+    if not yaml_info_path.exists():
+        err_str = f"The YAML file {yaml_info_path} doesn't exist."
+        raise ValueError(err_str)
+
+    with open(yaml_info_path, 'r') as f:
+        info = yaml.safe_load(f)
+        for key in split_method_path[1:]:
+            info = info[key]
```

Comment:
If the requested method is not in the yaml database file, this fails with a cryptic `KeyError`. I think this should be explicitly checked and a more user-friendly error generated.

Reply:
Yep, I completely agree with this point. Raising an error would help methods developers who have maybe missed adding in an entry when they have added in a new method, and would help users if they have accidentally made a mistake in the YAML (though ideally, some kind of validation of the YAML file being given to httomo will be done in the future to handle this user-error case).